Ponder application logging errors

### Version

0.9.10

### Current behavior

After a service restart, ponder logs print more that 100% progress and negative time to complete:

```ruby
4:44:14 AM INFO  app        Indexed 10 events with 130.2% complete and -29648ms remaining
4:44:22 AM INFO  app        Indexed 1667 events with 130.2% complete and -33145ms remaining
4:44:27 AM INFO  app        Indexed 1072 events with 130.2% complete and -35371ms remaining
4:44:28 AM INFO  app        Indexed 257 events with 130.2% complete and -35985ms remaining
4:44:36 AM INFO  app        Indexed 1667 events with 130.2% complete and -39496ms remaining
4:44:41 AM INFO  app        Indexed 1027 events with 130.2% complete and -41837ms remaining
4:44:41 AM INFO  app        Indexed 71 events with 130.2% complete and -42089ms remaining
```

### Expected behavior

Regardless service restarts, ponder logs print as accurate estimates as possible and a positive number for time to complete:

```ruby
4:42:49 AM INFO  app        Indexed 697 events with 65.1% complete and 10h 20m 48s remaining
4:42:51 AM INFO  app        Indexed 492 events with 65.1% complete and 10h 20m 47s remaining
4:42:54 AM INFO  app        Indexed 659 events with 65.1% complete and 10h 20m 48s remaining
4:42:55 AM INFO  app        Indexed 342 events with 65.1% complete and 10h 20m 47s remaining
4:42:58 AM INFO  app        Indexed 553 events with 65.1% complete and 10h 20m 46s remaining
```

### Steps to reproduce

<details><summary>Can only share the logs:</summary>

```ruby
4:42:42 AM INFO  app        Indexed 998 events with 65.1% complete and 10h 20m 47s remaining
4:42:46 AM INFO  app        Indexed 831 events with 65.1% complete and 10h 20m 47s remaining
4:42:49 AM INFO  app        Indexed 697 events with 65.1% complete and 10h 20m 48s remaining
4:42:51 AM INFO  app        Indexed 492 events with 65.1% complete and 10h 20m 47s remaining
4:42:54 AM INFO  app        Indexed 659 events with 65.1% complete and 10h 20m 48s remaining
4:42:55 AM INFO  app        Indexed 342 events with 65.1% complete and 10h 20m 47s remaining
4:42:58 AM INFO  app        Indexed 553 events with 65.1% complete and 10h 20m 46s remaining
4:43:00 AM ERROR indexing   Error while processing '/eth/Registry:NewOwner' event in '1' block 14744006
TypeError: fetch failed
    at Module.labelByHash (/app/apps/ensnode/src/lib/graphnode-helpers.ts:29:24)
    at /app/apps/ensnode/src/handlers/Registry.ts:133:31
  27 |  **/
  28 | export async function labelByHash(labelhash: Labelhash): Promise<string | null> {
> 29 |   const healResponse = await ensRainbowApiClient.heal(labelhash);
     |                        ^
  30 |
  31 |   if (healResponse.status === StatusCode.Success) {
  32 |     // original label found for the labelhash
caused by: Error: read ECONNRESET
    at TLSWrap.onStreamRead (node:internal/stream_base_commons:216:20)
Event arguments:
  node    0x93cdeb708b7545dc668eb9280176169d1c33cfd8ed6f04690a0bcc88a93fc4ae
  label   0xfc95e739c774165e2ed54462779d4a57ad8ed9fc040bbcb49ae1ba77d75354a8
  owner   0x283Af0B28c62C092C9727F1Ee09c02CA627EB7F5
Log:
  index     153
  address   0x00000000000C2E074eC69A0dFb2997BA6C7d2e1e
Transaction:
  hash   0x7aaec1fe42cdf43037fd83ebe48eee07b9b40e214ea437f751e532f079944de8
  from   0xe551BbdDd9be03d6f11aee4E354e875f0133CBef
  to     0x283Af0B28c62C092C9727F1Ee09c02CA627EB7F5
Block:
  hash        0x6d9db8f9366fe17c3e410bdd42c7fed05b0f9158735362034df8527adeae4222
  number      14744006
  timestamp   1652121939
4:43:00 AM WARN  process    Encountered indexing error, starting shutdown sequence
/app/apps/ensnode:
 ERR_PNPM_RECURSIVE_RUN_FIRST_FAIL  ensnode@0.0.1 start: `ponder start "--schema" "933bf48f-2307-4702-8b19-ba73f8a42e87"`
Exit status 1
> ensnode@0.0.1 start /app/apps/ensnode
> ponder start "--schema" "933bf48f-2307-4702-8b19-ba73f8a42e87"
4:43:08 AM INFO  build      Using Postgres database postgres.railway.internal:5432/railway (from DATABASE_URL env var)
4:43:09 AM INFO  database   Using database schema '933bf48f-2307-4702-8b19-ba73f8a42e87'
4:43:10 AM INFO  database   Dropped index 'multicoin_addr_changed_resolver_id_id_index' in schema '933bf48f-2307-4702-8b19-ba73f8a42e87'
4:43:10 AM INFO  database   Dropped index 'name_changed_resolver_id_id_index' in schema '933bf48f-2307-4702-8b19-ba73f8a42e87'
4:43:10 AM INFO  database   Dropped index 'abi_changed_resolver_id_id_index' in schema '933bf48f-2307-4702-8b19-ba73f8a42e87'
4:43:10 AM INFO  database   Dropped index 'pubkey_changed_resolver_id_id_index' in schema '933bf48f-2307-4702-8b19-ba73f8a42e87'
4:43:10 AM INFO  database   Dropped index 'text_changed_resolver_id_id_index' in schema '933bf48f-2307-4702-8b19-ba73f8a42e87'
4:43:10 AM INFO  database   Dropped index 'contenthash_changed_resolver_id_id_index' in schema '933bf48f-2307-4702-8b19-ba73f8a42e87'
4:43:10 AM INFO  database   Dropped index 'interface_changed_resolver_id_id_index' in schema '933bf48f-2307-4702-8b19-ba73f8a42e87'
4:43:10 AM INFO  database   Dropped index 'authorisation_changed_resolver_id_id_index' in schema '933bf48f-2307-4702-8b19-ba73f8a42e87'
4:43:10 AM INFO  database   Dropped index 'version_changed_resolver_id_id_index' in schema '933bf48f-2307-4702-8b19-ba73f8a42e87'
4:43:10 AM INFO  database   Reverted 0 unfinalized operations from 'domains'
4:43:10 AM INFO  database   Reverted 0 unfinalized operations from 'accounts'
4:43:10 AM INFO  database   Reverted 0 unfinalized operations from 'resolvers'
4:43:10 AM INFO  database   Reverted 0 unfinalized operations from 'registrations'
4:43:10 AM INFO  database   Reverted 0 unfinalized operations from 'wrapped_domains'
4:43:10 AM INFO  database   Reverted 0 unfinalized operations from 'transfers'
4:43:10 AM INFO  database   Reverted 0 unfinalized operations from 'new_owners'
4:43:10 AM INFO  database   Reverted 0 unfinalized operations from 'new_resolvers'
4:43:10 AM INFO  database   Reverted 0 unfinalized operations from 'new_ttls'
4:43:10 AM INFO  database   Reverted 0 unfinalized operations from 'wrapped_transfers'
4:43:10 AM INFO  database   Reverted 0 unfinalized operations from 'name_wrapped'
4:43:10 AM INFO  database   Reverted 0 unfinalized operations from 'name_unwrapped'
4:43:10 AM INFO  database   Reverted 0 unfinalized operations from 'fuses_set'
4:43:10 AM INFO  database   Reverted 0 unfinalized operations from 'expiry_extended'
4:43:10 AM INFO  database   Reverted 0 unfinalized operations from 'name_registered'
4:43:10 AM INFO  database   Reverted 0 unfinalized operations from 'name_renewed'
4:43:10 AM INFO  database   Reverted 0 unfinalized operations from 'name_transferred'
4:43:10 AM INFO  database   Reverted 0 unfinalized operations from 'addr_changed'
4:43:10 AM INFO  database   Reverted 0 unfinalized operations from 'multicoin_addr_changed'
4:43:10 AM INFO  database   Reverted 0 unfinalized operations from 'name_changed'
4:43:10 AM INFO  database   Reverted 0 unfinalized operations from 'abi_changed'
4:43:10 AM INFO  database   Reverted 0 unfinalized operations from 'pubkey_changed'
4:43:10 AM INFO  database   Reverted 0 unfinalized operations from 'text_changed'
4:43:10 AM INFO  database   Reverted 0 unfinalized operations from 'contenthash_changed'
4:43:10 AM INFO  database   Reverted 0 unfinalized operations from 'interface_changed'
4:43:10 AM INFO  database   Reverted 0 unfinalized operations from 'authorisation_changed'
4:43:10 AM INFO  database   Reverted 0 unfinalized operations from 'version_changed'
4:43:10 AM INFO  sync       Started '59144' historical sync with 99.7% cached
4:43:10 AM INFO  sync       Skipped '1' historical sync because all blocks are cached
4:43:10 AM INFO  sync       Started '8453' historical sync with 99.6% cached
4:43:12 AM INFO  sync       Completed '59144' historical sync
4:43:12 AM INFO  app        Indexed 322 events with 130.2% complete and -805ms remaining
4:43:13 AM INFO  app        Indexed 223 events with 130.2% complete and -1291ms remaining
4:43:14 AM INFO  app        Indexed 102 events with 130.2% complete and -1636ms remaining
4:43:16 AM INFO  app        Indexed 214 events with 130.2% complete and -2407ms remaining
4:43:16 AM INFO  sync       Completed '8453' historical sync
4:43:16 AM INFO  app        Indexed 134 events with 130.2% complete and -2717ms remaining
4:43:17 AM INFO  app        Indexed 190 events with 130.2% complete and -3122ms remaining
4:43:19 AM INFO  app        Indexed 345 events with 130.2% complete and -3820ms remaining
4:43:21 AM INFO  app        Indexed 580 events with 130.2% complete and -4845ms remaining
4:43:24 AM INFO  app        Indexed 702 events with 130.2% complete and -6147ms remaining
4:43:26 AM INFO  app        Indexed 699 events with 130.2% complete and -7381ms remaining
4:43:29 AM INFO  app        Indexed 779 events with 130.2% complete and -8762ms remaining
4:43:33 AM INFO  app        Indexed 1118 events with 130.2% complete and -10685ms remaining
```
</details>

### Link to repository

https://github.com/namehash/ensnode/tree/c119d06

### Anything else?

Feel free to check metrics if helpful:
- https://alpha.ensnode.io/metrics

--- a/.changeset/shaggy-actors-rhyme.md
+++ b/.changeset/shaggy-actors-rhyme.md
@@ -1,0 +1,5 @@
+---
+"ponder": patch
+---
+
+Fixed indexing metrics for apps with crash recovery.

--- a/.changeset/violet-points-enjoy.md
+++ b/.changeset/violet-points-enjoy.md
@@ -1,0 +1,5 @@
+---
+"ponder": patch
+---
+
+Fixed `/status` response.

--- a/packages/core/src/bin/utils/run.ts
+++ b/packages/core/src/bin/utils/run.ts
@@ -171,7 +171,8 @@ export async function run({
                   { network: network.name },
                   Math.max(
                     checkpoint.blockTimestamp -
-                      sync.seconds[network.name]!.start,
+                      sync.seconds[network.name]!.start -
+                      sync.seconds[network.name]!.cached,
                     0,
                   ),
                 );
@@ -185,7 +186,8 @@ export async function run({
                     { network: network.name },
                     Math.max(
                       checkpoint.blockTimestamp -
-                        sync.seconds[network.name]!.start,
+                        sync.seconds[network.name]!.start -
+                        sync.seconds[network.name]!.cached,
                       0,
                     ),
                   );
@@ -265,7 +267,9 @@ export async function run({
     common.metrics.ponder_historical_completed_indexing_seconds.set(
       label,
       Math.max(
-        sync.seconds[network.name]!.end - sync.seconds[network.name]!.start,
+        sync.seconds[network.name]!.end -
+          sync.seconds[network.name]!.start -
+          sync.seconds[network.name]!.cached,
         0,
       ),
     );

--- a/packages/core/src/sync/index.ts
+++ b/packages/core/src/sync/index.ts
@@ -263,17 +263,8 @@ export const createSync = async (params: {
 
   const updateHistoricalStatus = ({
     events,
-    checkpoint,
     network,
-  }: { events: Event[]; checkpoint: string; network: Network }) => {
-    if (Number(decodeCheckpoint(checkpoint).chainId) === network.chainId) {
-      status[network.name]!.block = {
-        timestamp: decodeCheckpoint(checkpoint).blockTimestamp,
-        number: Number(decodeCheckpoint(checkpoint).blockNumber),
-      };
-      return;
-    }
-
+  }: { events: Event[]; network: Network }) => {
     let i = events.length - 1;
     while (i >= 0) {
       const event = events[i]!;
@@ -388,16 +379,18 @@ export const createSync = async (params: {
           service: "sync",
           msg: `Sequenced ${events.length} '${network.name}' events for timestamp range [${decodeCheckpoint(cursor).blockTimestamp}, ${decodeCheckpoint(checkpoint).blockTimestamp}]`,
         });
+        updateHistoricalStatus({ events, network });
       } else {
         params.common.logger.debug({
           service: "sync",
           msg: `Sequenced ${events.length} events for timestamp range [${decodeCheckpoint(cursor).blockTimestamp}, ${decodeCheckpoint(checkpoint).blockTimestamp}]`,
         });
+
+        for (const network of params.indexingBuild.networks) {
+          updateHistoricalStatus({ events, network });
+        }
       }
 
-      for (const network of params.indexingBuild.networks) {
-        updateHistoricalStatus({ events, checkpoint, network });
-      }
       yield events;
       cursor = checkpoint;
     }
@@ -663,11 +656,7 @@ export const createSync = async (params: {
           });
 
           if (to < from) {
-            params.onRealtimeEvent({
-              type: "reorg",
-              checkpoint: to,
-              network,
-            });
+            params.onRealtimeEvent({ type: "reorg", checkpoint: to, network });
           }
         }
 

--- a/packages/core/src/sync/index.ts
+++ b/packages/core/src/sync/index.ts
@@ -806,7 +806,13 @@ export const createSync = async (params: {
             getOmnichainCheckpoint({ tag: "finalized" }),
           ),
         ).blockTimestamp,
-        cached: decodeCheckpoint(params.initialCheckpoint).blockTimestamp,
+        cached: decodeCheckpoint(
+          min(
+            getOmnichainCheckpoint({ tag: "end" }),
+            getOmnichainCheckpoint({ tag: "finalized" }),
+            params.initialCheckpoint,
+          ),
+        ).blockTimestamp,
       };
     }
   } else {
@@ -820,7 +826,13 @@ export const createSync = async (params: {
             getOmnichainCheckpoint({ tag: "finalized" }),
           ),
         ).blockTimestamp,
-        cached: decodeCheckpoint(params.initialCheckpoint).blockTimestamp,
+        cached: decodeCheckpoint(
+          min(
+            getOmnichainCheckpoint({ tag: "end" }),
+            getOmnichainCheckpoint({ tag: "finalized" }),
+            params.initialCheckpoint,
+          ),
+        ).blockTimestamp,
       };
     }
   }

